WordPress/Classes/Services/MediaSettings.swift

```diff
@@ -1,13 +1,95 @@
 import Foundation
+import AVFoundation
 
 class MediaSettings: NSObject {
     // MARK: - Constants
     fileprivate let maxImageSizeKey = "SavedMaxImageSizeSetting"
     fileprivate let removeLocationKey = "SavedRemoveLocationSetting"
+    fileprivate let maxVideoSizeKey = "SavedMaxVideoSizeSetting"
+
 
     fileprivate let minImageDimension = 150
     fileprivate let maxImageDimension = 3000
 
+    enum VideoResolution: String {
+        case size640x480 = "AVAssetExportPreset640x480"
+        case size960x540 = "AVAssetExportPreset960x540"
+        case size1280x720 = "AVAssetExportPreset1280x720"
+        case size1920x1080 = "AVAssetExportPreset1920x1080"
+        case size3840x2160 = "AVAssetExportPreset3840x2160"
+        case sizeOriginal = "AVAssetExportPresetPassthrough"
+
+        var videoPreset: String {
+            switch (self) {
+            case(.size640x480):
+                return AVAssetExportPreset640x480
+            case(.size960x540):
+                return AVAssetExportPreset960x540
+            case(.size1280x720):
+                return AVAssetExportPreset1280x720
+            case(.size1920x1080):
+                return AVAssetExportPreset1920x1080
+            case(.size3840x2160):
+                return AVAssetExportPreset3840x2160
+            case(.sizeOriginal):
+                return AVAssetExportPresetPassthrough
+            }
+        }
+
+        var description: String {
+            switch (self) {
+            case(.size640x480):
+                return NSLocalizedString("VGA (640x480)", comment: "Indicates a video will be resized to 640x480 when uploaded.")
```

Comment:
I wonder if we could also ditch 540p – it's only marginally more than 480p, and it's less common than either 480 or 720.